gui/dash.py

```diff
@@ -1,17 +1,70 @@
+import os
+import subprocess
+import sys
+
 import PyQt5.QtWidgets as qtw
 
 from .input_pane import InputPane
 from .output_pane import OutputPane
 
+__location__ = os.path.join(os.getcwd(), os.path.dirname(__file__))
+__parent_dir__ = os.path.realpath(os.path.join(__location__, ".."))
+
 class Dash(qtw.QWidget):
 
     def __init__(self):
         super().__init__()
 
-        self._input_pane = InputPane()
+        self._input_pane = InputPane(self._on_button_press)
         self._output_pane = OutputPane()
 
         self._grid = qtw.QGridLayout()
         self._grid.addWidget(self._input_pane, 0, 0)
         self._grid.addWidget(self._output_pane, 0, 1)
         self.setLayout(self._grid)
+
+    def _on_button_press(self):
+        py = sys.executable
+        parsing_script = os.path.join(__parent_dir__, "line_trimmer.py")
+        in_file = self._input_pane.get_input_path()
+        out_file = self._input_pane.get_output_path()
+
+        try:
+            self._validate_inputs(in_file, out_file)
+        except Exception as e:
+            self._output_pane.setPlainText(e.__str__())
+            return
+     
+        process = subprocess.run([py, parsing_script, in_file, out_file], stdout=subprocess.PIPE)
```

Comment:
might be better to run in debug mode here ?

Reply:
fixed